File: 프로그래머스/Python/l2_기능개발.py
# https://programmers.co.kr/learn/courses/30/lessons/42586
# 프로그래머스 > 스택/큐
# 기능개발 (Level 2)

# import math

# def solution(p, s):
#     answer= []
#     time = 0
#     cnt = 0
    
#     while p:
#         if (p[0]+time*s[0]) >=100:
#             p.pop(0)
#             s.pop(0)
#             cnt +=1
#         else:
#             if cnt > 0:
#                 answer.append(cnt)
#                 cnt =0
#             time +=1
#     answer.append(cnt)
#     return answer

def solution(p, s):
    answer = []
    time = 0
    now = 0

    while now < len(p):
        cnt = 0
        # 속도 더하기
        for i in range(now, len(p)):
            p[i] += s[i]
        
        for i in range(now, len(p)):
            if p[now] >=100:
                cnt +=1
                now = i+1
        
        if cnt!=0:
            answer.append(cnt)

        time +=1
        
            
    print(answer)   
    return answer


solution([93, 30, 55], 	[1, 30, 5])
solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1])

File: 프로그래머스/JAVA/l2_기능개발.java
import java.util.ArrayList;
import java.util.Arrays;

public class l2_기능개발 {
    public static void main(String[] args){
        int[] progresses = {93,30,55};
        int[] speeds = {1,30,5};

        int[] answer = solution(progresses, speeds);
    }
    public static int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> result = new ArrayList<>();

        int now = 0;
        int cnt = 0;

        while (now < progresses.length){
            cnt = 0;

            for (int i =0; i<progresses.length;i++){
                progresses[i] += speeds[i];
            }
            for (int i=now; i<progresses.length;i++){
                if (progresses[now]>=100){
                    cnt +=1;
                    now = i+1;
                }
            }
            if (cnt!=0){
                result.add(cnt);
            }
        }
        int[] answer = new int[result.size()];
        for (int i =0; i<result.size(); i++){
            answer[i] = result.get(i).intValue();
        }
        System.out.println(Arrays.toString(answer));
        return answer;
    }
}
